core/sdk-core/src/main/java/software/amazon/awssdk/core/client/builder/SdkDefaultClientBuilder.java

```diff
@@ -252,7 +253,8 @@ protected SdkClientConfiguration mergeChildDefaults(SdkClientConfiguration confi
      * Apply global default configuration
      */
     private SdkClientConfiguration mergeGlobalDefaults(SdkClientConfiguration configuration) {
-        Supplier<ProfileFile> defaultProfileFileSupplier = new Lazy<>(ProfileFile::defaultProfileFile)::getValue;
+        Supplier<ProfileFileSupplier> profileFileSupplierSupplier = new Lazy<>(ProfileFileSupplier::defaultSupplier)::getValue;
```

Comment:
nit : can we name this something like profileFileSupplierProvider instead of the double "Supplier" ?

Reply:
renamed the variable